src/main/java/org/roaringbitmap/RunContainer.java

```diff
@@ -942,6 +932,40 @@ public boolean equals(Object o) {
     return false;
   }
 
+  private boolean equals(RunContainer runContainer) {
+    if (runContainer.nbrruns != this.nbrruns) {
+      return false;
+    }
+    for (int i = 0; i < nbrruns; ++i) {
+      if (getValue(i) != runContainer.getValue(i)) {
+        return false;
+      }
+      if (getLength(i) != runContainer.getLength(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean equals(ArrayContainer arrayContainer) {
+    if (getCardinality() != arrayContainer.getCardinality()) {
```

Comment:
Given that we need to traverse the RunContainer to compute its cardinality, this might have a higher cost than checking the cardinality in the loop below, I guess...